lib/cylc/batch_sys_manager.py

```diff
@@ -695,6 +700,10 @@ def _job_submit_impl(
             # batch_sys.submit should handle OSError, if relevant.
             ret_code, out, err = batch_sys.submit(job_file_path)
         else:
+            env = None
+            if hasattr(batch_sys, "SUBMIT_CMD_ENV"):
+                env = dict(os.environ)
```

Comment:
I suppose the job submit command is executed in this environment anyway.  We're simply overriding any variables that are also in SUBMIT_CMD_ENV, and setting the rest to the values they already have anyway, so it doesn't matter ... is that your reasoning?   However, isn't it still the case that loading `os.environ` wouldn't be necessary at all except to make your test work?